src/sentry/integrations/slack/action_endpoint.py

```diff
@@ -0,0 +1,245 @@
+from sentry.api.base import Endpoint
+from sentry.models import Group, Integration, Project, IdentityProvider, Identity
+from sentry.utils import json
+from sentry.utils.http import absolute_uri
+
+from .utils import build_attachment, logger
+
+LINK_IDENTITY_MESSAGE = "Looks like you haven't linked your Sentry account with your Slack identity yet! <{link_url}|Link your identity now> to perform actions in Sentry through Slack."
+
+RESOLVE_SELECTOR = {
+    'label': 'Resolve issue',
+    'type': 'select',
+    'name': 'resolve_type',
+    'placeholder': 'Select the resolution target',
+    'value': 'resolved',
+    'options': [
+        {
+            'label': 'Immediately',
+            'value': 'resolved'
+        },
+        {
+            'label': 'In the next release',
+            'value': 'resolved:inNextRelease'
+        },
+        {
+            'label': 'In the current release',
+            'value': 'resolved:inCurrentRelease'
+        },
+    ],
+}
+
+
+class SlackActionEndpoint(Endpoint):
+    authentication_classes = ()
+    permission_classes = ()
+
+    def on_assign(self, request, identity, group, action):
+        assignee = action['selected_options'][0]['value']
+
+        if assignee == 'none':
+            assignee = None
+
+        try:
+            self.update_group(group, identity, {'assignedTo': assignee})
+        except client.ApiError as e:
+            logger.error('slack.action.sentry-client-error', extra={'error': e.body})
+
+    def on_status(self, request, identity, group, action, data, integration):
+        status = action['value']
+
+        status_data = status.split(':', 1)
+        status = {'status': status_data[0]}
+
+        resolve_type = status_data[-1]
+
+        if resolve_type == 'inNextRelease':
+            status.update({'statusDetails': {'inNextRelease': True}})
+        elif resolve_type == 'inCurrentRelease':
+            status.update({'statusDetails': {'inRelease': 'latest'}})
+
+        try:
+            self.update_group(group, identity, status)
+        except client.ApiError as e:
+            logger.error('slack.action.sentry-client-error', extra={'error': e.body})
+
+    def update_group(self, group, identity, data):
+        return client.put(
+            path='/projects/{}/{}/issues/'.format(
+                group.project.organization.slug,
+                group.project.slug,
+            ),
+            params={'id': group.id},
+            data=data,
+            user=identity.user
+        )
+
+    def open_resolve_dialog(self, data, group, integration):
+        # XXX(epurkhiser): In order to update the original message we have to
+        # keep track of the response_url in the callback_id. Definitely hacky,
+        # but seems like there's no other solutions [1]:
+        #
+        # [1]: https://stackoverflow.com/questions/46629852/update-a-bot-message-after-responding-to-a-slack-dialog#comment80795670_46629852
+        callback_id = json.dumps({
```

Comment:
Calling this out again here in the review.. Slack provides no easy way to get the message object that triggered a dialog interaction, so we have to serialize enough information into the only field we control (the `callback_id`) just to be keep a reference to be able to update the message.